src/form-builder/DateField.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { useField, FieldHookConfig } from 'formik';
+
+import { FieldProps } from './types';
+import { getMessage } from '../utils/i18n';
+import {
+  ValidationFunction,
+  chainValidations,
+  required,
+} from '../utils/validation';
+
+import { SimpleDate } from '@department-of-veterans-affairs/component-library/Date';
+
+export type ComponentLibraryDatePart = {
+  value: string;
+  dirty: boolean;
+};
+
+/**
+ * What the Date in the component library expects.
+ */
+export type ComponentLibraryDateValue = {
+  day: ComponentLibraryDatePart;
+  month: ComponentLibraryDatePart;
+  year: ComponentLibraryDatePart;
+};
+
+export type DateProps = FieldProps<string>;
+
+/**
+ * Convert the date string passed to DateField to the object expected by the
+ * component library Date component.
+ */
+const dateStringToValue = (dateString = '') => {
+  // TODO: Make this conversion handle edge cases
```

Comment:
Yeah, I'm _super_ wary of using JavaScript's Date because it's...[unreliable](https://stackoverflow.com/questions/2488313/javascripts-getdate-returns-wrong-date). The short version is that it'll behave differently depending on the user's location (setting) and browser's implementation. This is part of the reason why `moment` and `date-fns` exist.

But yeah, eventually I'd like to parse this using a much more comprehensive library like `date-fns`. I'd trust that a lot more.

Reply:
Bummer :/